presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PruneTableScanColumns.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.sql.planner.DependencyExtractor;
+import com.facebook.presto.sql.planner.PlanNodeIdAllocator;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.SymbolAllocator;
+import com.facebook.presto.sql.planner.iterative.Lookup;
+import com.facebook.presto.sql.planner.iterative.Rule;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.planner.plan.ProjectNode;
+import com.facebook.presto.sql.planner.plan.TableScanNode;
+import com.google.common.collect.ImmutableList;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class PruneTableScanColumns
+        implements Rule
+{
+    @Override
+    public Optional<PlanNode> apply(PlanNode node, Lookup lookup, PlanNodeIdAllocator idAllocator, SymbolAllocator symbolAllocator)
+    {
+        if (!(node instanceof ProjectNode)) {
+            return Optional.empty();
+        }
+
+        ProjectNode parent = (ProjectNode) node;
+
+        PlanNode source = lookup.resolve(parent.getSource());
+        if (!(source instanceof TableScanNode)) {
+            return Optional.empty();
+        }
+
+        TableScanNode child = (TableScanNode) source;
+
+        Set<Symbol> dependencies = DependencyExtractor.extractUnique(parent.getAssignments().getExpressions());
```

Comment:
Why doesn't this need to be like `VALUES`?
```java
// we need to compute the intersection in case some dependencies are symbols from
// the outer scope (i.e., correlated queries)
Set<Symbol> dependencies = Sets.intersection(DependencyExtractor.extractUnique(parent.getAssignments().getExpressions()), childOutputs);
```